prelude/ts/src/sk_types.ts

```diff
@@ -196,6 +196,35 @@ export interface WasmSupplier {
   completeWasm: (wasm: object, utils: Utils) => void;
 }
 
+function utf8Encode(string) {
+  const utf8 = new Array();
+
+  for (let i = 0; i < string.length; i++) {
+    let charcode = string.charCodeAt(i);
+
+    if (charcode < 0x80) {
+      utf8.push(charcode);
+    } else if (charcode < 0x800) {
+      utf8.push((charcode >> 6) | 0xc0);
+      utf8.push((charcode & 0x3f) | 0x80);
+    } else if (charcode < 0xd800 || charcode >= 0xe000) {
+      utf8.push((charcode >> 12) | 0xe0);
+      utf8.push(((charcode >> 6) & 0x3f) | 0x80);
+      utf8.push((charcode & 0x3f) | 0x80);
+    } else {
+      // surrogate pair
+      i++;
+      charcode = 0x10000 + (((charcode & 0x3ff) << 10) | (string.charCodeAt(i) & 0x3ff));
+      utf8.push((charcode >> 18) | 0xf0);
+      utf8.push(((charcode >> 12) & 0x3f) | 0x80);
+      utf8.push(((charcode >> 6) & 0x3f) | 0x80);
+      utf8.push((charcode & 0x3f) | 0x80);
+    }
+  }
+
+  return utf8;
+}
```

Comment:
new TextEncoder().encode(string)

Reply:
TextEncoder is a large dependency, that's why I did it this way.
I don't know ...

Reply:
We use it for the protocol e.g.[1]. It's built in to browsers since ~2014. https://developer.mozilla.org/en-US/docs/Web/API/TextEncoder

I just figure that it's more likely to handle any weird edges and it'll probably be in C so faster. Oh and every little helps with minimizing JS code size.

[1]: https://github.com/SkipLabs/skdb/blob/main/sql/ts/src/skdb_orchestration.ts#L86

Reply:
Let me try to revisit this. might take a bit of time because I am having issues with my setup.

Reply:
Ok so looking closer, encode is not what we need. It returns an Uint8Array which is not what we need here. We need the unicode int value for each character. I am going to check if there is something built-in that does that.

Reply:
Ok, so I found a way to do this natively. Diff coming up!